### PR TITLE
WIP: first steps towards refactoring for tests v2

### LINKED ISSUE #1163: Tests for Supporting files
Add browser tests for the supporting file uploads

### PR DESCRIPTION
**Pull request by jyenes**
Background
Started adding tests for multiple file uploads

Any relevant tickets
First step for #1163

How has this been tested?
ITS ALL TESTS

🍌 💃

### CODE REVIEW

File: server/xpub-server/entities/manuscript/resolvers/removeSupportingFiles.test.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+      const fakeManuscript = {
+        filename: 'manuscript.pdf',
+        type: 'MANUSCRIPT_SOURCE',
+        url: 'manuscript/',
+      }
+      const manuscript = new Manuscript({
+        createdBy: userId,
+        files: [fakeManuscript],
+      })
+      await expectRemoveSupportingFilesDoesNothing(manuscript, userId)
```
Comment (diversemix) [inline server/xpub-server/entities/manuscript/resolvers/removeSupportingFiles.test.js]: rewrite to arrange>act>assert paradigm

File: server/xpub-server/entities/manuscript/resolvers/removeSupportingFiles.test.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
+const { createTables } = require('@pubsweet/db-manager')
```
Comment (diversemix) [inline server/xpub-server/entities/manuscript/resolvers/removeSupportingFiles.test.js]: Move test into controller

Comment (diversemix) [reply server/xpub-server/entities/manuscript/resolvers/removeSupportingFiles.test.js]: Maybe move S3Storage into controller

File: server/xpub-server/entities/manuscript/resolvers/removeUploadedManuscript.js [hunk 3]
```diff
@@ -11,6 +11,7 @@ async function removeUploadedManuscript(_, vars, { user }) {
 
   if (manuscriptUploadIndex > -1) {
     manuscript.files.splice(manuscriptUploadIndex, 1)
+    // TODO: Should we remove the content from S3 ???
```
Comment (diversemix) [inline server/xpub-server/entities/manuscript/resolvers/removeUploadedManuscript.js]: If we do overwrite the exiting manuscript in s3 - then we need to test that it does

File: server/xpub-server/index.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
 const { merge } = require('lodash')
 const fs = require('fs')
+const S3Storage = require('./s3Storage')
```
Comment (diversemix) [inline server/xpub-server/index.js]: move s3Storage into the controller

File: server/xpub-server/s3Storage.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  static async putContent(fileObject, content, { size } = {}) {
+    if (!fileObject.id) {
+      throw new Error('File has no ID, must be saved first')
+    }
+
+    return s3
+      .putObject({
+        Body: content,
+        Key: `${fileObject.url}/${fileObject.id}`,
+        ContentType: fileObject.mimetype,
```
[the remark this answers is not in the document]
Comment (damnhipster) [reply server/xpub-server/s3Storage.js]: #1224

File: server/xpub-server/entities/manuscript/resolvers/removeUploadedManuscript.js (hunk 3 above)
Comment (damnhipster) [reply server/xpub-server/entities/manuscript/resolvers/removeUploadedManuscript.js]: #1163

**Comment by diversemix** [comment]
@damnhipster .... this should go green now, if it does we will merge and take the comments onto the next PR for this issue

**Pull request merged**